composer/trainer/trainer.py

```diff
@@ -90,6 +90,8 @@ class contains metrics relevant to the specific dataset. Set to ``None`` for no
             or 'amp' (recommended). (default: ``Precision.FP32``).
         scale_schedule_ratio (float, optional): Ratio by which to scale the training duration and learning rate
             schedules. See :func:`scale_schedule` for details. (default: ``1.0``)
+        adaptive_grad_accum (boolean, optional): Dynamically scale down minibatch size and use gradient 
+            accumulation if train_batch_size is too large for GPU. (default: ``True``)
```

Comment:
I think I'd prefer for this behavior to be gated behind the `grad_accum` parameter. The behavior could be something like "If -1, set `grad_accum` to the per-device batch size and enable dynamic adjustments.